scripts/pipeline_generator/pipeline_generator.py

```diff
@@ -0,0 +1,51 @@
+import os
+import re
+from typing import List
+
+from pydantic import BaseModel, field_validator
+
+
+class PipelineGeneratorConfig(BaseModel):
+    run_all: bool
+    list_file_diff: List[str]
+    container_registry: str
+    container_registry_repo: str
+    commit: str
+    test_path: str  # List of tests
+    external_hardware_test_path: str  # List of external hardware tests
+    pipeline_file_path: str  # Path to the output pipeline file
+
+    @property
+    def container_image(self) -> str:
+        return f"{self.container_registry}/{self.container_registry_repo}:{self.commit}"
+
+    @field_validator("test_path")
+    @classmethod
+    def check_test_path(cls, value: str) -> str:
+        if not os.path.exists(value):
+            raise ValueError(f"Test path {value} does not exist")
```

Comment:
field validator is supposed to be checking if the values of the fields are "legal", and they are supposed to be pure functions. it is part of the marshal / unmarshaling operation of a data structure, so it cannot have dependencies on the file system.

for example, if one marshals an object into bytes, it should always be able to unmarshal the bytes back to the object, no matter what the state of the file system is (or if it even has a file system).

please check against the file system in another explicit function.

Reply:
I removed these validators for now.. will add them into helper later on